### PR TITLE
Add git_info module; Include create release GitHub Actions; Include PyPI publishing GitHub Actions

### LINKED ISSUE #2: Add git_info module
We will migrate existing copies of `git_info` module from ReQUIAM and LD-Cool-P over.

- [x] Include `git_info` module
- [x] Update python-package.yml to CI testing
- [x] Remove `os` library and only use `pathlib`﻿
- [x] Ensure that an import of package will import all codes (this is for packaging)

### LINKED ISSUE #3: Add create release GitHub Actions
This follows our previous software where we trigger on a version tag pushed
﻿

### LINKED ISSUE #6: Include GitHub Actions for automated PyPI publishing
This package should be published to PyPI to aid in easy `pip install` by specifying it in the requirements.txt for other ReDATA related repositories

A few things:
First, `redata` is not taken on PyPI, so I am thinking of having it package as such
 - [ ] `redata_commons` should be renamed to `redata` for packaging purposes
 - [x] The commons code need to be in a sub-folder, as a sub-package
 - [x] We will want a GitHub action workflow called "pypi-publish.yml" to publish on tagged release
 - [ ] Testing will be needed of the .tar.gz and .whl file to ensure that packaging is done properly
 - [x] We will use the API token approach as that provides greater control on scope. Temporary tokens have been added as secrets

~Finally, I'm not sure if we want to rename the repository to `redata` from `redata_commons`. Not sure if we want that namespace for something else.~ This has been decided (item list amended above).

### PR DESCRIPTION
**Pull request by astrochun**
Closes #2
Closes #3
Closes #6 

- [x] Bump version from v0.0.1 -> v0.1.0
- [x] Add minimal CHANGELOG file
- [x] Update README.md with more information
- [x] Add docstrings for `GitInfo`
- [x] Add create release GitHub Action #3 
- [x] Add `git_info` module
- [x] Add PyPI publishing GitHub actions #6 
- [x] Add PyPI API token secrets (@astrochun) [Note: token will need to change with scope limited to `redata` package] #6

Commit history:
- Add git_info module
- Add __init__.py for redata_commons
- Add unit tests for git_info module
- Add setup.py script for packaging
- Add python-package GitHub Actions
- Add create-release.yml #3 [ci skip]
- Exclude PyCharm from .gitignore
- git_info: Remove os library usage
- Add MANIFEST.in [ci skip]

### CODE REVIEW

**Review by astrochun — COMMENTED** [review]
Changes look good and has been tested in 3.7 dev env and with CI.

Will need to bump version to `v0.1.0`

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install pytest pytest-cov
+        python setup.py install
+    - name: Test with pytest
+      run: |
+        pytest --cov-report term-missing --cov-config=.coveragerc --cov=redata_commons tests
```
Comment (damian-romero) [inline .github/workflows/python-package.yml:45]: I believe `redata_commons` here should change to `redata`
```suggestion
        pytest --cov-report term-missing --cov-config=.coveragerc --cov=redata tests
```

**Review by damian-romero — CHANGES_REQUESTED** [review]
I was able to build locally. Except for a minor fix, everything looks good. I was wondering if we should test it on `https://test.pypi.org/`

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (astrochun) [reply .github/workflows/python-package.yml:45]: Ah yes, I renamed and forgot to update the GitHub action.

**Comment by astrochun** [comment]
> I was able to build locally. Except for a minor fix, everything looks good. I was wondering if we should test it on `https://test.pypi.org/`

What was the minor fix that was needed?

Yes, we can publish say v0.0.1 to test PyPI. Note that versioning is fix on PyPI.  Once you publish you can't re-publish with the same version. It's not like GitHub where you can delete a tag to re-tag it. The final version of this release is 0.1.0 so that won't be an issue here. You can `pip install --index-url https://test.pypi.org/simple/ redata` to use Test PyPI. Once I have it published I'll note it in the PR.

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (damian-romero) [reply .github/workflows/python-package.yml:45]: @astrochun this was the minor fix I talked about in my review. Next time I'll reference things like this for clarity.

**Comment by damian-romero** [comment]
> Yes, we can publish say v0.0.1 to test PyPI. Note that versioning is fix on PyPI. Once you publish you can't re-publish with the same version. It's not like GitHub where you can delete a tag to re-tag it. The final version of this release is 0.1.0 so that won't be an issue here. You can `pip install --index-url https://test.pypi.org/simple/ redata=0.1.0` to use Test PyPI. Once I have it published I'll note it in the PR.

Sounds good. I'll be on the lookout for when you want me to merge this into main. If I understand correctly, this repository will be using GitHub flow instead of git flow.

**Comment by astrochun** [comment]
I did publish v0.0.1 to Test PyPI: https://test.pypi.org/project/redata/

`pip install -i https://test.pypi.org/simple/ redata==0.0.1` should work

**Review by damian-romero — APPROVED** [review]

**Review by astrochun — COMMENTED** [review]
Changes look good

**Pull request merged**